clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/names/GoogleLabeledResourceNamer.java

```diff
@@ -99,9 +99,6 @@ private static String getClusterName(String app, String stack, String detail) {
 
   private static <T> void setIfPresent(Consumer<T> setter, T value) {
     if (value != null) {
-      if (value.equals("")) {
-        value = null;
-      }
```

Comment:
It is actually this way on purpose. The use case is if you have some legacy infrastructure that is named in a way that Frigga parses as something, you can overwrite that part by setting an empty label. Without this feature, you can only overwrite it with another value, not remove/ignore it. E.g. if you have a legacy instance group named `myApp-foo-bar` and you want it to appear in Spinnaker as `myApp-foo`, you can do that by setting an empty `spinnaker-moniker-detail` label.

Reply:
I think you could omit/delete the `details` moniker label (while including the others) to remove the detail from the legacy server group.

Reply:
Yeah. I think this block was something I should have removed earlier anyway and never should have ended up in the PR. Thanks for fixing :)